### ISSUE TITLE
assert_equal and other assertions for MiniTest don't appear in dropdown

### ISSUE THREAD

**Issue by garyking** [castwide/solargraph#43]
Generally speaking, variable and function auto-completes just fine, by appearing in the dropdown.

But I'm using MiniTest and assertions in my unit tests, and typing `ass` doesn't show `assert_equal` and other assertions that I've used in the same block, in the dropdown.

Some basic test code:

```ruby
require 'test_helper'

class TestCore < Minitest::Test
  def test_core
    assert_equal 'a', core('a')
  end
end
```

Of course, it can be hard for Solargraph to find where the function was defined, I imagine. I require `test_helper`, which is a `test/test_helper.rb` file that contains:

```ruby
require 'minitest/autorun'
require 'minitest/reporters'

reporter_options = { color: true, slow_count: 3 }
Minitest::Reporters.use!([Minitest::Reporters::DefaultReporter.new(
  reporter_options,
)])
```

And the assertions come from there.

**Comment by castwide** [comment]
It's a little tricky to map DSLs provided by libraries like MiniTest, RSpec, and Sinatra, but I'm working on a feature that might help.

If you've looked at the default .solargraph.yml generated by `solargraph config`, you might have noticed a section called `domains`. That's an experimental option for mapping a namespace's methods as a DSL. It's inactive at the moment, but I'm in the process of migrating it to the LSP. This example would expose Sinatra's DSL in completion:

```yaml
domains:
- Class<Sinatra::Application>
```

The downside to this feature is that it enables the DSL for all files in the workspace. I plan to add a macro directive that activates it on a per-file basis. In a test script, you would add something like this:

```ruby
# Just an example. I'm not sure what the MiniTest DSL's namespace is
# @!domain Minitest::Test
require 'test_helper'
# ...
```

Hopefully that will help fill in some of the methods that Solargraph can't locate through static analysis.

This feature should be enabled in an upcoming version (maybe 0.20.0, but I'm not sure yet).

**Comment by castwide** [comment]
Most of the functionality already existed, so it was fairly easy to implement a proof of concept. Here's an example that imports the RSpec DSL:

![image](https://user-images.githubusercontent.com/3704922/38843850-e93701e4-41be-11e8-9a80-99da5afb1879.png)

It'll be an experimental feature of v0.20.0.

**Comment by castwide** [comment]
Support for the `@!domain` directive is published in v0.20.0. Example using Minitest:

![image](https://user-images.githubusercontent.com/3704922/39116220-de72c0e8-46b1-11e8-828f-d47eefea3fcf.png)

**Comment by castwide** [comment]
The `@!domain` directive at least partially resolves this issue. If anyone has recommendations for improving DSL support, please open a new issue.
